### PR TITLE
Add Datadog to regression tests 

### PR DESCRIPTION
**Pull request by markus-hinsche**
**Proposed changes**:
- Send `test_run_time`, `train_run_time`, and `total_run_time` to Datadog via datadog API
- Visualize on a [Datadog dashboard](https://app.datadoghq.eu/dashboard/xsh-dvk-jca/mhinsches-dashboard-tue-nov-9-101519-am?from_ts=1636462443929&to_ts=1636466043929&live=true)
- enable for CPU and GPU

**Questions for reviewers:**
- I would like to know a good location in the codebase where to put the function (could be a new module) `transform_to_seconds`
- After this I would like to add unit tests for at least this function
- Do you suggest to add documentation for the feature? Where can I put this?

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by markus-hinsche** [comment]
/modeltest

```yml
include:
 - dataset: ["Carbon Bot"]
   config: ["Sparse + DIET(bow) + ResponseSelector(bow)"]
```

File: .github/scripts/mr_publish_results.py [hunk 1]
```diff
@@ -22,6 +25,66 @@
...
+    options = {
+        "statsd_host": "127.0.0.1",
+        "statsd_port": 8125,
+        "statsd_constant_tags": tags_list,
+    }
+    initialize(**options)
+
+    # Send  metrics
+    metrics = {
+        "test_run_time": os.environ["TEST_RUN_TIME"],
```
Comment (tczekajlo) [inline .github/scripts/mr_publish_results.py:79]: Please use prefix for metrics, e.g. `rasa.perf.benchmark.test_run_time`

File: .github/workflows/ci-model-regression.yml [hunk 2]
```diff
@@ -604,6 +622,10 @@ jobs:
           DATASET_REPOSITORY_BRANCH: ${{ matrix.dataset_branch }}
           TYPE: ${{ matrix.type }}
           DATASET_COMMIT: ${{ steps.set-dataset-commit.outputs.dataset_commit }}
+          BRANCH: ${{ steps.extract_branch.outputs.branch }}
+          DD_ENV: regression-tests # ${{ matrix.test }}
```
Comment (koernerfelicia) [inline .github/workflows/ci-model-regression.yml:626]: How come matrix.test is commented?

**Comment by koernerfelicia** [comment]
> I would like to know a good location in the codebase where to put the function 

I think it could go in `rasa.utils.common`, but double-checking on this one (might be we don't want it in the codebase proper), I will get back to you! 

> Do you suggest to add documentation for the feature? Where can I put this?

I would say no. If it does go into the codebase it will need a docstring.

File: .github/scripts/mr_publish_results.py [hunk 3]
```diff
@@ -22,6 +26,66 @@
...
+        "accelerator_type": os.environ["ACCELERATOR_TYPE"],
+        "github_run_id": os.environ["GITHUB_RUN_ID"],
+        "github_sha": os.environ["GITHUB_SHA"],
+        "github_event": os.environ["GITHUB_EVENT_NAME"],
+        "type": os.environ["TYPE"],
+        "branch": os.environ["BRANCH"],
+    }
+    tags_list = [f"{k}:{v}" for k, v in tags.items()]
+    options = {
+        "statsd_host": "127.0.0.1",
```
Comment (tczekajlo) [inline .github/scripts/mr_publish_results.py:72]: why do you need it for? I assume you're using a DD agent to send metrics, it'll be better if you send the metrics via API. https://docs.datadoghq.com/api/latest/metrics/#submit-metrics

File: .github/workflows/ci-model-regression.yml (hunk 2 above)
Comment (tczekajlo) [reply .github/workflows/ci-model-regression.yml:626]: It looks like it's copy & pastes from the other workflow where `${{ matrix.text }}` makes sense. Here a constant name is ok.

I'd change the name to `rasa-regression-tests`.

File: .github/workflows/ci-model-regression.yml [hunk 4]
```diff
@@ -544,12 +550,19 @@ jobs:
...
+
       - name: Run test
         id: run_test
         if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
         env:
           TFHUB_CACHE_DIR: ~/.tfhub_cache/
           OMP_NUM_THREADS: 1
+          DD_ENV: regression-tests
+          DD_SERVICE: rasa
+          DD_ARGS: --ddtrace
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression.yml:565]: It's not needed, it works only for pytests.

File: .github/workflows/ci-model-regression.yml [hunk 5]
```diff
@@ -561,21 +574,21 @@ jobs:
           fi
 
           if [[ "${{ matrix.type }}" == "nlu" ]]; then
-            poetry run rasa train nlu --quiet -u ${DATASET_DIR}/${DATASET}/${TRAIN_DIR} -c dataset/configs/${CONFIG} --out ${DATASET_DIR}/models/${DATASET}/${CONFIG}
+            poetry run ddtrace-run rasa train nlu --quiet -u ${DATASET_DIR}/${DATASET}/${TRAIN_DIR} -c dataset/configs/${CONFIG} --out ${DATASET_DIR}/models/${DATASET}/${CONFIG}
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression.yml:577]: Looking at the data that we can get from APM, there are no benefits of using trace + APM in this particular case. I'd remove ddtrace.

File: .github/workflows/ci-model-regression.yml [hunk 6]
```diff
@@ -418,6 +418,12 @@ jobs:
     if: "!contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:model-regression-tests') && needs.read_test_configuration.outputs.configuration_id != ''"
 
     steps:
+      - name: Run DataDog Agent
+        run: |
+          docker run -d --name dd_agent -p 8125:8125/udp -p 8126:8126 -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY=${{ secrets.DD_API_KEY }} -e DD_SITE="datadoghq.eu" -e DD_DOGSTATSD_NON_LOCAL_TRAFFIC=true -e "DD_INSIDE_CI=true" -e "DD_HOSTNAME=none" -e GITHUB_ACTIONS=true -e CI=true gcr.io/datadoghq/agent:7
```
Comment (tczekajlo) [inline .github/workflows/ci-model-regression.yml:423]: If we send metrics via API/DataDog client, and we removed ddtrace, using the agent is not needed then.

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-model-regression.yml (hunk 5 above)
Comment (tczekajlo) [reply .github/workflows/ci-model-regression.yml:577]: Or at least I wouldn't enable it by default, rather by an extra-label.

**Comment by koernerfelicia** [comment]
> I would like to know a good location in the codebase where to put the function

Okay, so I think we would not put this into the codebase, but rather leave it where it is. You could maybe separate it out into its own file for other GHA to use, but I'd leave that decision up to infra (@tczekajlo ?)

**Comment by tczekajlo** [comment]
> > I would like to know a good location in the codebase where to put the function
> 
> Okay, so I think we would not put this into the codebase, but rather leave it where it is. You could maybe separate it out into its own file for other GHA to use, but I'd leave that decision up to infra (@tczekajlo ?)

Let's leave where it is 😄

**Comment by markus-hinsche** [comment]
/modeltest

```yml
include:
 - dataset: ["Carbon Bot"]
   config: ["Sparse + DIET(bow) + ResponseSelector(bow)"]
```

**Review by tczekajlo — APPROVED** [review]
LGTM ⭐

**Comment by markus-hinsche** [comment]
@tczekajlo Sorry for adding commits after your LGTM. I just had the idea to enable the functionality also for GPU, this could work now since we don't need the docker dd_agent container anymore

**Review by koernerfelicia — APPROVED** [review]

**Pull request merged**